src/app/clusters/mode-base-server/mode-base-server.h

```diff
@@ -0,0 +1,215 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "mode-base-cluster-objects.h"
+#include <app/AttributeAccessInterface.h>
+#include <app/AttributePersistenceProvider.h>
+#include <app/CommandHandlerInterface.h>
+#include <app/util/af.h>
+#include <map>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ModeBase {
+
+class Instance : public CommandHandlerInterface, public AttributeAccessInterface
+{
+public:
+    /**
+     * This is a helper function to build a mode option structure. It takes the label/name of the mode,
```

Comment:
This helper function is:

1. Not really public API; it's a utility meant for subclasses, right?
2. An attractive nuisance: it's only useful if you store all your stuff in memory all the time, but it makes it look like that's the intended way to do things, whereas it's just the way that was easy in our example app.

I would suggest removing it from here altogether.  If we want to use it in the all-clusters-app, we can create an intermediate class there that it lives on, with clear documentation about how this only makes sense if all your mode state is in memory all the time.

Reply:
I removed it completely [here](https://github.com/project-chip/connectedhomeip/pull/27504/commits/a70bfcaa7a0f626b3df3c2345997cf5b50302f61).